src/librustc/Cargo.toml

```diff
@@ -13,12 +13,12 @@ arena = { path = "../libarena" }
 fmt_macros = { path = "../libfmt_macros" }
 graphviz = { path = "../libgraphviz" }
 log = "0.3"
+owning_ref = "0.3.3"
```

Comment:
This is fine to pull in owning_ref under MIT. We have historically accepted minor dependencies under permissive licenses including MIT and BSD, as these are maximally compatible with Rust's other licenses.

There's a tidy check that tries to keep our licenses in order but it may have some gaps yet.

Thanks for pinging me @nikomatsakis